docs/source/en/main_classes/deepspeed.mdx

```diff
@@ -2259,6 +2259,14 @@ rank1:
 
 This was a very basic example and you will want to adapt it to your needs.
 
+### `generate` nuances
+
+When using multiple GPUs with ZeRO Stage-3, one has to synchronize the GPUs by calling `generate(..., synced_gpus=True)`. If this is not done if one GPU finished generating before other GPUs the whole system will hang as the rest of the GPUs will not be able to received the shard of weights from the GPU that stopped generating.
+
+Starting from `transformers>4.27.2`, if `synced_gpus` isn't set to a non-`None` value, it'll be set to `True` automatically if these conditions are detected. But you can still override the value of `synced_gpus` if need to.
```

Comment:
```suggestion
Starting from `transformers>=4.28`, if `synced_gpus` isn't passed along, it'll be set to `True` automatically if these conditions are detected. But you can still override the value of `synced_gpus` if need to.
```
Nit on the version, another patch is coming soon and there might be others before 4.28. None will contain this since we only cherry-pick commit that fix regressions for patch releases.

Reply:
check on the version, but I don't think "passed along" is a very clear definition.

would this work?

> if `synced_gpus` isn't explicitly specified

Reply:
I like the first one better.